### DISCUSSION TITLE
Invalid SSL Certificate

### DISCUSSION THREAD

**Discussion by flow-motion** [akeeba/panopticon#597] [Q&A (Community Support)]
Hey Everyone  
One of my pages is marked in the List of panopticon with a lock and when i hover that lock it says: Invalid SSL certificate
I checked my hoster Panel and ssl is active 
I also checked my joomla and it is set to force ssl - whole page. 

What can be the problem?

Have a great time, 
Florian

**Comment by nikosdion** [comment]
Check the information on the SSL certificate. Does it indicate an expiration, or that it is about to expire? You get a notification before it expires, you can set up exactly when. If you are using Let's Encrypt the certificate is typically only updated 1-2 days in advance of its expiration, so you may want to set the warning time to just 1 day in this case.

**Reply by rkonikpl** [reply]
I have a similar problem. In my case, it is related to one hosting provider. I have already spoken with them and everything seems correct.

What exactly is the issue? When I go to a specific page, there is a padlock in the top right corner. It is red. When I click on it, I get the message:

Invalid or self-signed certificate
The signature of your SSL/TLS certificate cannot be verified against the Trusted Certificate Authorities known on the server. This usually means that you are using a self-signed certificate, which will be rejected by browsers.

However, when I visit the site:
https://www.sslshopper.com/ssl-checker.html#hostname=dkdietetyka.pl

Everything is correct.

This only happens with one hosting provider. What is going on?

**Reply by nikosdion** [reply]
Do you have any local caches? I visited your site and I get a valid certificate. If you, too, now get a valid certificate it means that your host only updates the TLS certificate after it expires which is, of course, a problem.

The correct way to deal with Let's Encrypt certificates is to try update them 3 to 7 days before their expiration, giving you time to address any problems that may arise, and prevent causing a situation where you serve an expired TLS certificate.

**Reply by rkonikpl** [reply]
I don't know if I explained it well? And did you understand me correctly?
When I log into Panopticon and select a page from the list by clicking on its title, I see a red padlock in the upper right corner after entering. When I click on it, I see an error message, as shown in the screenshot.
All browsers indicate that everything is fine. Various online tools also show that everything is okay. Only Panopticon displays an error. I am trying to resolve what the cause is, whether it is the fault of Panopticon or the hosting. I don't want to blame either side, but simply find the cause and solution.

![ssl](https://github.com/akeeba/panopticon/assets/2079176/8a829313-1bf5-40be-906d-b5161abfc975)

**Reply by brianteeman** [reply]
@nikodsion could this be a problem with the ca bundle? The error message points in that direction

**Reply by nikosdion** [reply]
Yes, this is a problem with the CA bundle on the server where Panopticon is hosted.

@rkonikpl Ask your host to make sure that they have installed an up-to-date CA bundle on their system, and that their php.ini points to it. The way this feature works is that it asks PHP to establish a TLS connection with your site, capturing the TLS certificate information. It then relays the TLS certificate information captured by PHP back to you.

**Reply by rkonikpl** [reply]
I'm starting to get confused. Certificates are not my strong suit. I received the following response from the hosting provider:

**Our certificates are correctly generated and support only browsers with SNI. If the software queries the site from browsers without SNI, it may cause issues with proper CA validation. Is the application configured to handle such certificates?**

I have translated the text from my native language, and I hope I didn't make any mistakes.

Does the question contribute to solving my problem, or is it a dead end?

**Reply by nikosdion** [reply]
Your host's answer is completely irrelevant to the question asked. They are talking about the TLS certificates they use on sites hosted with them. The question was about the **Certification Authorities (CA) cache bundle configured on the server itself**. One has nothing to do with the other. Their reply is about how requests from the Internet to the server are handled, what we asked is about how requests from the server to the Internet are handled.
 
As your host already knows, every version of PHP from 5.6 onwards does support SNI. Kind reminder, PHP 5.6 was released a decade ago, on August 28th, 2014. Panopticon requires PHP 8.1 or later. Moreover, it is besides the point as we can clearly see that the correct certificate is returned. So, yes, obviously, SNI is used.

The question I told you to ask them is **on the server where Panopticon runs** have they made sure they have an up-to-date Certification Authorities (CA) bundle, and have they configured PHP properly so as to see this CA bundle? This is basic systems administration. If they can't do that, run away!
